test/integration/clusterthrottle_test.go

```diff
@@ -163,6 +164,57 @@ var _ = Describe("Clusterthrottle Test", func() {
 		})
 	})
 
+	When("Group Pods", func() {
+		var (
+			podPassedGroup    []*corev1.Pod
+			thr               *v1alpha1.ClusterThrottle
+			podThrottledGroup []*corev1.Pod
+		)
+		BeforeEach(func() {
+			thr = MustCreateClusterThrottle(ctx,
+				MakeClusterThrottle(throttleName).Selector(DefaultNs, throttleKey, throttleName).
+					ThresholdPod(4).
+					ThresholdCpu("4").
+					Obj(),
+			)
+
+			for i := 0; i < 2; i++ {
+				podPassedGroup = append(podPassedGroup, MustCreatePod(ctx, MakePod(DefaultNs, fmt.Sprintf("passed-pod%d", i), "100m").Annotation(groupNameAnnotation, "passed").Label(throttleKey, throttleName).Obj()))
+			}
+			for i := 0; i < 3; i++ {
+				podThrottledGroup = append(podThrottledGroup, MustCreatePod(ctx, MakePod(DefaultNs, fmt.Sprintf("throttled-pod%d", i), "100m").Annotation(groupNameAnnotation, "throttled").Label(throttleKey, throttleName).Obj()))
+			}
+		})
+		It("should not schedule podThrottledGroup", func() {
+			Eventually(AsyncAll(
+				WakeupBackoffPod(ctx),
+				ClusterThottleHasStatus(
+					ctx, thr.Name,
+					ClthrOpts.WithCalculatedThreshold(thr.Spec.Threshold),
+					ClthrOpts.WithUsedPod(len(podPassedGroup)),
+					ClthrOpts.WithUsedCpuReq(fmt.Sprintf("%dm", len(podPassedGroup)*100)),
+					ClthrOpts.WithPodThrottled(false),
+					ClthrOpts.WithCpuThrottled(false),
+				),
+				func(g types.Gomega) {
+					for _, pod := range podPassedGroup {
+						PodIsScheduled(ctx, DefaultNs, pod.Name)
```

Comment:
Does this line work?
Returned value from the `PodIsScheduled` seems not to be used 🤔 

Same for `MustPodFailedScheduling`, `PodIsScheduled` below.